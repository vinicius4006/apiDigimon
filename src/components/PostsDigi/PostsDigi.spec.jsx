import { render, screen } from '@testing-library/react';
import { PostsDigi } from '.';

const props = {
    
    digiList: [
        {
            
            name: 'nameDigimon',
            img: 'img/digimon.png',
            level: 'levelDigimon',
        },
        {
            
            name: 'nameDigimon 2',
            img: 'img/digimon2.png',
            level: 'levelDigimon 2',
        },
        {
            
            name: 'nameDigimon 3',
            img: 'img/digimon3.png',
            level: 'levelDigimon 3',
        },
    ]

};

describe('<PostsDigi />', () => {
    it('should render posts', () => {
        render(<PostsDigi {...props}/>)


        expect(screen.getAllByRole('heading', { name: /nameDigimon/i }))
            .toHaveLength(3);

        expect(screen.getAllByRole('img', { name: /nameDigimon/i }))
            .toHaveLength(3);

        expect(screen.getAllByText(/levelDigimon/i))
            .toHaveLength(3);

        expect(screen.getByRole('img', { name: /nameDigimon 3/i }))
            .toHaveAttribute('src', 'img/digimon3.png');


    });

    it('should match snapshot', () => {
        const {container} = render(<PostsDigi {...props}/>)

        expect(container.firstChild).toMatchSnapshot();
        

    });

    it('should not render posts', () => {
        render(<PostsDigi/>)

        expect(screen.queryByRole('heading', { name: /nameDigimon/i }))
            .not.toBeInTheDocument();
        

    });
});