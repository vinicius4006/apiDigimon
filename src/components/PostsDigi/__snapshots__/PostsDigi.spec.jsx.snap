// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsDigi /> should match snapshot 1`] = `
<div
  class="digiList"
>
  <div
    class="digiPost"
  >
    <img
      alt="nameDigimon"
      src="img/digimon.png"
    />
    <div
      class="digi-content"
    >
      <h1
        class="h1-content"
      >
        nameDigimon
      </h1>
      <p>
        levelDigimon
      </p>
    </div>
  </div>
  <div
    class="digiPost"
  >
    <img
      alt="nameDigimon 2"
      src="img/digimon2.png"
    />
    <div
      class="digi-content"
    >
      <h1
        class="h1-content"
      >
        nameDigimon 2
      </h1>
      <p>
        levelDigimon 2
      </p>
    </div>
  </div>
  <div
    class="digiPost"
  >
    <img
      alt="nameDigimon 3"
      src="img/digimon3.png"
    />
    <div
      class="digi-content"
    >
      <h1
        class="h1-content"
      >
        nameDigimon 3
      </h1>
      <p>
        levelDigimon 3
      </p>
    </div>
  </div>
</div>
`;

exports[`<PostsDigi /> should not render posts 1`] = `
<div
  class="digiList"
>
  <div
    class="digiPost"
  >
    <img
      alt="nameDigimon"
      src="img/digimon.png"
    />
    <div
      class="digi-content"
    >
      <h1
        class="h1-content"
      >
        nameDigimon
      </h1>
      <p>
        levelDigimon
      </p>
    </div>
  </div>
  <div
    class="digiPost"
  >
    <img
      alt="nameDigimon 2"
      src="img/digimon2.png"
    />
    <div
      class="digi-content"
    >
      <h1
        class="h1-content"
      >
        nameDigimon 2
      </h1>
      <p>
        levelDigimon 2
      </p>
    </div>
  </div>
  <div
    class="digiPost"
  >
    <img
      alt="nameDigimon 3"
      src="img/digimon3.png"
    />
    <div
      class="digi-content"
    >
      <h1
        class="h1-content"
      >
        nameDigimon 3
      </h1>
      <p>
        levelDigimon 3
      </p>
    </div>
  </div>
</div>
`;
